File: semester.py
#!/usr/bin/python

import sys, math, time

def week(start):
	return math.floor((time.time()-start)/(60*60*24*7))

def percent(start, end):
	return round((time.time()-start)/(end-start)*100, 1)

if sys.argv[1] == "-w" or sys.argv[1] == "--week":
	print "Week number is " + str(week(int(sys.argv[2])))
elif sys.argv[1] == "-p" or sys.argv[1] == "--percent":
	print str(percent(int(sys.argv[2]), int(sys.argv[3]))) + "% of the semester is spend"
else:
	print "Sorry, command not found !"

File: semester.c
#include <stdio.h>
#include <math.h>
#include <time.h>

int week(int now, int start)
{
	return floor((now-start)/(60*60*24*7));
}

int percent(int now, int start, int end)
{
	return ((now-start)/(end-start)*100);
}

int main(int argc, char *argv[])
{
	time_t now = time (NULL);

	if (strcmp(argv[1], "-w", 2) == 0)
		printf("Week number is %d\n", week(now, (int)strtoul(argv[2], 0, 0)));
	else if (strcmp(argv[1], "-p", 2) == 0)
		printf("%d%s of the semester is spend\n", percent(now, (int)strtoul(argv[2], 0, 0), (int)strtoul(argv[3], 0, 0)), "%");
	else
		printf("Sorry, command not found !\n");
}
